test/index.js

```diff
@@ -62,6 +62,13 @@ async function loadDeviceFactory(deviceKind) {
 async function createDeviceInstance(deviceKind, factory) {
     if (factory.metadata.auth.type === 'none')
         return new factory(mockEngine, { kind: deviceKind });
+    if (factory.metadata.auth.type === 'basic') {
+        // credentials are stored in test/[DEVICE ID].cred.json
+        const credentialsPath = path.resolve(path.dirname(module.filename), deviceKind + '.cred.json');
+        const args = require(credentialsPath);
```

Comment:
if you use `require` you don't need the full path, as `require` resolves the path relative to the calling module by itself.

Reply:
Understand. I'll simplify it.